src/main/kotlin/day05/Wand.kt

```
@@ -0,0 +1,29 @@
+    _name: String?,
+    _power: Double,
+) {
+    var name: String? = if(_name == null || _name.length < 3) throw IllegalArgumentException() else _name // 이야 컴파일러가 똑똑해서 여기서 이미 null값 걸러지는 줄 알고서는 setter에서 타입을 바꿔버리네!
+        set(value: String?) {                                                                             // 그럼 아예 내가 명시적으로 name의 타입을 String?로 바꿔버리지 뭐
+            if (value == null || value.length < 3) { // 왜 always false야 윗 문장은 생성자의 매개변수를 집어넣어주는거고 set은 이후 수정시 집어넣어주는건데
+                throw IllegalArgumentException()
+            }
+            field = value
+        }
+
+    var power = if(_power < 0.5 || _power > 100.0) throw IllegalArgumentException() else _power
```

detekt (1.23.7)

[warning] A call to the default constructor of an exception was detected. Instead one of the constructor overloads should be called. This allows to provide more meaningful exceptions.

(detekt.exceptions.ThrowingExceptionsWithoutMessageOrCause)